mailbox/api/src/test/java/org/apache/james/mailbox/events/EventDeadLettersContract.java

```diff
@@ -115,68 +118,77 @@ static Event event(Event.EventId eventId) {
         default void storeShouldThrowWhenNullGroup() {
             EventDeadLetters eventDeadLetters = eventDeadLetters();
 
-            assertThatThrownBy(() -> eventDeadLetters.store(NULL_GROUP, EVENT_1))
+            assertThatThrownBy(() -> eventDeadLetters.store(NULL_GROUP, EVENT_1, INSERTION_ID_1))
                 .isInstanceOf(IllegalArgumentException.class);
         }
 
         @Test
         default void storeShouldThrowWhenNullEvent() {
             EventDeadLetters eventDeadLetters = eventDeadLetters();
 
-            assertThatThrownBy(() -> eventDeadLetters.store(GROUP_A, NULL_EVENT))
+            assertThatThrownBy(() -> eventDeadLetters.store(GROUP_A, NULL_EVENT, INSERTION_ID_1))
                 .isInstanceOf(IllegalArgumentException.class);
         }
 
         @Test
-        default void storeShouldThrowWhenBothGroupAndEventAreNull() {
+        default void storeShouldThrowWhenNullInsertionId() {
             EventDeadLetters eventDeadLetters = eventDeadLetters();
 
-            assertThatThrownBy(() -> eventDeadLetters.store(NULL_GROUP, NULL_EVENT))
+            assertThatThrownBy(() -> eventDeadLetters.store(GROUP_A, EVENT_1, NULL_INSERTION_ID))
                 .isInstanceOf(IllegalArgumentException.class);
         }
 
         @Test
-        default void storeShouldStoreGroupWithCorrespondingEvent() {
+        default void storeShouldThrowWhenBothGroupAndEventAreNull() {
             EventDeadLetters eventDeadLetters = eventDeadLetters();
 
-            eventDeadLetters.store(GROUP_A, EVENT_1).block();
-            assertThat(eventDeadLetters.failedEvent(GROUP_A, EVENT_1.getEventId()).block())
-                .isEqualTo(EVENT_1);
+            assertThatThrownBy(() -> eventDeadLetters.store(NULL_GROUP, NULL_EVENT, NULL_INSERTION_ID))
+                .isInstanceOf(IllegalArgumentException.class);
         }
 
         @Test
-        default void storeShouldIgnoreStoreDuplicatedEventsPerGroup() {
+        default void storeShouldStoreGroupWithCorrespondingEvent() {
             EventDeadLetters eventDeadLetters = eventDeadLetters();
 
-            eventDeadLetters.store(GROUP_A, EVENT_1).block();
-            eventDeadLetters.store(GROUP_A, EVENT_1).block();
-            eventDeadLetters.store(GROUP_A, EVENT_1).block();
-
-            assertThat(eventDeadLetters.failedEventIds(GROUP_A).toStream())
-                .containsExactly(EVENT_ID_1);
+            eventDeadLetters.store(GROUP_A, EVENT_1, INSERTION_ID_1).block();
+            assertThat(eventDeadLetters.failedEvent(GROUP_A, INSERTION_ID_1).block())
+                .isEqualTo(EVENT_1);
         }
 
+//        @Test
+//        default void storeShouldIgnoreStoreDuplicatedEventsPerGroup() {
```

Comment:
We need to handle this commented test.

Maybe we can be answering the question `what happens if I store 2 time the same event with a different insertionId?`

Then we might expect the event to be duplicated.